### PR TITLE
New Resource: `azurerm_api_management_product_tag` - Support API Management product tag

### PR DESCRIPTION
**Pull request by xuzhang3**
Support assign tag API Management product
```log
=== RUN   TestAccApiManagementProductTag_basic
=== PAUSE TestAccApiManagementProductTag_basic
=== RUN   TestAccApiManagementProductTag_requiresImport
=== PAUSE TestAccApiManagementProductTag_requiresImport
=== CONT  TestAccApiManagementProductTag_basic
=== CONT  TestAccApiManagementProductTag_requiresImport
--- PASS: TestAccApiManagementProductTag_basic (527.23s)
--- PASS: TestAccApiManagementProductTag_requiresImport (557.47s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement 578.725s
```

### CODE REVIEW

File: website/docs/r/api_management_product_tag.html.markdown [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+The following arguments are supported:
+
+* `api_management_name` - (Required) The name of the API Management Service. Changing this forces a new resource to be created.
+
+* `product_name` - (Required) The name of the API Management product. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group in which the API Management Service should be exist. Changing this forces a new resource to be created.
+
+* `tag_name` - (Required) The name which should be used for this API Management Tag. Changing this forces a new API Management Tag to be created.
```
Comment (katbyte) [inline website/docs/r/api_management_product_tag.html.markdown:65]: as the resource name is tag should this be
```suggestion
* `name` - (Required) The name which should be used for this API Management Tag. Changing this forces a new API Management Tag to be created.
```

so its `azurerm_api_management_product_tag.r.name` not `azurerm_api_management_product_tag.r.tag_name` ?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/apimanagement/api_management_product_tag_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  published             = false
+}
+
+resource "azurerm_api_management_tag" "test" {
+  api_management_id = azurerm_api_management.test.id
+  name              = "acctestTag-%d"
+}
+
+resource "azurerm_api_management_product_tag" "test" {
+  product_name        = azurerm_api_management_product.test.product_id
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_product_tag_resource_test.go:102]: this is called product_name but take product id?

File: website/docs/r/api_management_product_tag.html.markdown [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `api_management_name` - (Required) The name of the API Management Service. Changing this forces a new resource to be created.
+
+* `product_name` - (Required) The name of the API Management product. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group in which the API Management Service should be exist. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/api_management_product_tag.html.markdown:63]: couldn't these all be replaced with product_id? `api_management_product_id = azurerm_api_management_product.test.product.id` ?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/apimanagement/api_management_product_tag_resource_test.go (hunk 2 above)
Comment (xuzhang3) [reply internal/services/apimanagement/api_management_product_tag_resource_test.go:102]: Yes, API doc named it `produceId`, to distinguish with the resource ID like `/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.ApiManagement/service/service1/products/product1/tags/tagId1`,  I think `product_name` is better here.

File: website/docs/r/api_management_product_tag.html.markdown (hunk 3 above)
Comment (xuzhang3) [reply website/docs/r/api_management_product_tag.html.markdown]: rename `product_id` to `api_management_product_id`

Comment (xuzhang3) [reply website/docs/r/api_management_product_tag.html.markdown]: `product_name` has be renamed to `api_management_product_id`

**Review by katbyte — APPROVED** [review]
LGTM ♻️

**Pull request merged**